Add pre-built Linux/WSL binary to release installer

## Feedback: Poor Linux/WSL Installation Experience

### Current Behavior

The install script (`install.sh`) currently exits with an error on Linux:

```
Error: No Linux binary available yet. Build from source:
  cargo install --git https://github.com/mgblackwater/zero-drift-chat
```

Following the suggested fallback (`cargo install --git ...`) on a standard **stable** Rust toolchain fails immediately with:

```
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> wacore/binary/src/lib.rs:1:1
  |
1 | #![feature(portable_simd)]

error[E0432]: unresolved import `core::simd::Select`
 --> wacore/binary/src/encoder.rs:3:5
```

The dependency `wacore-binary` (from `jlucaso1/whatsapp-rust`) uses nightly-only Rust features (`portable_simd`), so installation silently requires **nightly Rust** — which is not documented anywhere.

### Steps to Reproduce

1. Run the installer on WSL2 / Ubuntu / any Linux distro with stable Rust
2. Follow the printed fallback command
3. Build fails

### Proposed Improvements

1. **Ship a pre-built Linux (`x86_64-unknown-linux-gnu`) binary** in GitHub Releases, and update `install.sh` to download it — same as the macOS/Windows flow. This is the highest-impact fix.

2. **If source-build remains the fallback**, update the error message and docs to explicitly state nightly Rust is required:
   ```
   Error: No Linux binary available yet. Build from source (requires nightly Rust):
     rustup toolchain install nightly
     cargo +nightly install --git https://github.com/mgblackwater/zero-drift-chat
   ```

3. **Consider pinning a `rust-toolchain.toml`** in the repo so `cargo install` automatically selects the correct toolchain without user intervention.

### Environment

- OS: WSL2 (Ubuntu, Linux 6.6.x)
- Rust stable: 1.x (build fails)
- Rust nightly: build succeeds after ~8 min compile

### Summary

A first-time user on Linux following the official install instructions will hit a silent, confusing build failure with no clear path forward. A pre-built Linux binary (or at minimum clearer error messaging + nightly requirement docs) would significantly improve the onboarding experience.

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -21,6 +21,10 @@ jobs:
             target: aarch64-apple-darwin
             artifact: zero-drift-chat-macos-aarch64
             binary: zero-drift-chat
+          - os: ubuntu-22.04
+            target: x86_64-unknown-linux-gnu
+            artifact: zero-drift-chat-linux-x86_64
+            binary: zero-drift-chat
 
     runs-on: ${{ matrix.os }}
 
@@ -65,3 +69,4 @@ jobs:
           files: |
             artifacts/zero-drift-chat-windows-x86_64.exe/zero-drift-chat-windows-x86_64.exe
             artifacts/zero-drift-chat-macos-aarch64/zero-drift-chat-macos-aarch64
+            artifacts/zero-drift-chat-linux-x86_64/zero-drift-chat-linux-x86_64

--- a/install.sh
+++ b/install.sh
@@ -15,9 +15,15 @@ case "$OS" in
     ASSET="zero-drift-chat-macos-aarch64"
     ;;
   Linux)
-    echo "Error: No Linux binary available yet. Build from source:"
-    echo "  cargo install --git https://github.com/$REPO"
-    exit 1
+    case "$ARCH" in
+      x86_64) ASSET="zero-drift-chat-linux-x86_64" ;;
+      *)
+        echo "Error: No Linux $ARCH binary available. Build from source (requires nightly Rust):"
+        echo "  rustup toolchain install nightly"
+        echo "  cargo +nightly install --git https://github.com/$REPO"
+        exit 1
+        ;;
+    esac
     ;;
   MINGW*|MSYS*|CYGWIN*)
     echo "On Windows, use PowerShell instead:"

